gradle-plugin/src/main/java/proguard/gradle/ProGuardTask.java

```diff
@@ -67,6 +68,13 @@ public abstract class ProGuardTask extends DefaultTask
     // INTERNAL USE ONLY - write extra data entries to this jar
     private File extraJar;
 
+    public ProGuardTask() {
+        if (GradleVersion.current().compareTo(GradleVersion.version("7.4")) >= 0) {
+            // This method was added in Gradle 7.4
+            notCompatibleWithConfigurationCache("https://github.com/Guardsquare/proguard/issues/254");
```

Comment:
This is more of a workaround than a full fix, but it saves users from having to do this manually while a proper fix is worked on.